test/AST-Quake/array.cpp

```diff
@@ -0,0 +1,80 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+// RUN: cudaq-quake %s | cudaq-opt | FileCheck %s
+
+#include <cudaq.h>
+
+struct T {
+   void operator()(int N) __qpu__ {
+      cudaq::qreg Q(N);
+      x(Q);
+   }
+};
+
+struct S {
+   void operator()() __qpu__ {
+      int arr[3] = {4, 8, 10};
+      T{}(arr[0]);
+      T{}(arr[1]);
+      T{}(arr[2]);
```

Comment:
I would also recommend adding a similar test with `std::vector<int>` just for completeness. It could even be in the same file and reuse `struct T`, perhaps something like this:
```suggestion
      int arr[3] = {4, 8, 10};
      T{}(arr[0]);
      T{}(arr[1]);
      T{}(arr[2]);
      std::vector<int> vec{1,2,3};
      T{}(vec[0]);
      T{}(vec[1]);
      T{}(vec[2]);
      for (auto i = 0; i < vec.size(); i++)
        T{}(vec[i]);
```

Reply:
Ok.